runtime/common/src/gateway.rs

```diff
@@ -12,26 +12,18 @@
 
 use cfg_primitives::AccountId;
 use pallet_evm::AddressMapping;
-use polkadot_parachain::primitives::Sibling;
+use polkadot_parachain_primitives::primitives::Sibling;
 use sp_core::{crypto::AccountId32, Get, H160};
 use sp_runtime::traits::AccountIdConversion;
 
-use crate::account_conversion::AccountConverter;
+use crate::account_conversion::AccountMapping;
 
-pub struct GatewayAccountProvider<T, XcmConverter>(core::marker::PhantomData<(T, XcmConverter)>);
+pub fn get_gateway_account<T: pallet_evm_chain_id::Config + parachain_info::Config>() -> AccountId {
+	let sender_account: AccountId =
+		Sibling::from(parachain_info::Pallet::<T>::get()).into_account_truncating();
```

Comment:
As a heads up: 

We should probably hardcode that one here to NOT use a dynamic `ParaId` but enforce it to use `2031`. The LP solidity side has a hardcode sending address, which works as we are having the same `ParaId` both on demo and on mainnet, so the LP logic works on both. BUT with the given swap to the new relay-chain we will get a new id breaking the LP part.

Reply:
So I would move this to `crate::types::ids` or so.

Reply:
I suppose LPs are never supposed to be used on Altair? Else we cannot hardcode the CFG para id.

Reply:
Why not? Is should not be important who the sender is. Just that it is fixed.

Reply:
Because Altair has a different para id